### ISSUE TITLE
Side effect hook on transform

### ISSUE THREAD

**Issue by Korijn** [pygfx/pygfx#571]
See #562

We should provide a hook that can be used to apply transformation side effects without causing a recursion stack overflow.

**Comment by Korijn** [comment]
Is this still a problem?

**Comment by panxinmiao** [comment]
Yes, the light helpers needs to update its state before every frame is rendered (to stay synchronized with the target light). This should be an automatic update process.

However, we currently do not have a good mechanism for automatic frame-by-frame updates. As a workaround, the light helpers updates its state within the callback method of the AffineTransform to achieve automatic updates. It's important to note that if the updated state involves any transform-related changes, it may recursively trigger the update method, resulting in infinite recursion (#562 was created to resolve this problem).

That said, this is not an ideal approach. Once we implement a proper frame-by-frame automatic update mechanism (which is also needed for other objects like Skeletons), we will no longer need this workaround. @almarklein seems to be working on or preparing to address this.

**Comment by almarklein** [comment]
Is this still an issue after the pylinalg and transform refactorings?
